### PR TITLE
Add java-lambda-kata-solutions module

### PR DESCRIPTION
**Pull request by prathasirisha**
Signed-off-by: Sirisha Pratha <sirisha.pratha@bnymellon.com>

### CODE REVIEW

File: java-lambda-kata-solutions/src/test/java/bnymellon/codekatas/lambdakata/jdk/FunctionalInterfaceTest.java [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+        Assert.assertEquals("UPPERCASE", toUppercase.apply("uppercase"));
+        List<String> lowercase = List.of("a", "b", "c", "d");
+        Set<String> uppercase = lowercase.stream().map(toUppercase).collect(Collectors.toSet());
+        Assert.assertEquals(Set.of("A", "B", "C", "D"), uppercase);
+    }
+
+    @Test
+    public void supplier()
+    {
+        // TODO - Convert this anonymous inner class to a lambda and then to a constructor reference
```
Comment (donraab) [inline java-lambda-kata-solutions/src/test/java/bnymellon/codekatas/lambdakata/jdk/FunctionalInterfaceTest.java:108]: please remove the TODO comment

File: java-lambda-kata-solutions/src/test/java/bnymellon/codekatas/lambdakata/jdk/FunctionalInterfaceTest.java [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+        Assert.assertEquals(Interval.evensFromTo(1, 10), evens);
+    }
+
+    @Test
+    public void predicateIsOdd()
+    {
+        var numbers = IntStream.rangeClosed(1, 10).boxed().collect(Collectors.toList());
+
+        Predicate<Integer> oddPredicate = integer -> integer % 2 == 1;
+
```
Comment (donraab) [inline java-lambda-kata-solutions/src/test/java/bnymellon/codekatas/lambdakata/jdk/FunctionalInterfaceTest.java:85]: please remove the extra blank line

File: java-lambda-kata-solutions/src/test/java/bnymellon/codekatas/lambdakata/jdk/FunctionalInterfaceTest.java [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+ *
+ * Then follow the TODOs in each test and convert the anonymous inner classes to lambdas and/or method references.
+ */
+public class FunctionalInterfaceTest
+{
+    @Test
+    public void consumer()
+    {
+        var strings = List.of("one", "two", "three");
+
```
Comment (donraab) [inline java-lambda-kata-solutions/src/test/java/bnymellon/codekatas/lambdakata/jdk/FunctionalInterfaceTest.java:54]: please remove the extra blank line

**Review by donraab — CHANGES_REQUESTED** [review]
Hi @prathasirisha, overall the module looks great! Thanks for the contribution! I requested a few small cosmetic changes. Thanks!

File: java-lambda-kata-solutions/src/test/java/bnymellon/codekatas/lambdakata/jdk/FunctionalInterfaceTest.java (hunk 3 above)
Comment (prathasirisha) [reply java-lambda-kata-solutions/src/test/java/bnymellon/codekatas/lambdakata/jdk/FunctionalInterfaceTest.java:54]: Removed.

File: java-lambda-kata-solutions/src/test/java/bnymellon/codekatas/lambdakata/jdk/FunctionalInterfaceTest.java (hunk 2 above)
Comment (prathasirisha) [reply java-lambda-kata-solutions/src/test/java/bnymellon/codekatas/lambdakata/jdk/FunctionalInterfaceTest.java]: Removed.

File: java-lambda-kata-solutions/src/test/java/bnymellon/codekatas/lambdakata/jdk/FunctionalInterfaceTest.java (hunk 1 above)
Comment (prathasirisha) [reply java-lambda-kata-solutions/src/test/java/bnymellon/codekatas/lambdakata/jdk/FunctionalInterfaceTest.java]: Removed.

**Review by donraab — APPROVED** [review]
LGTM.

**Pull request merged**